### DISCUSSION TITLE
Data import

### DISCUSSION THREAD

**Discussion by Duvuki** [graphia-app/graphia#37] [Q&A] [answered]
I am trying to import data in cvs format. Csv contains source and target column, type: undirected, and weight:1 (for all rows) While importing Graphia reports error Empty graph. Could you provide more info on data preparation part? I also tried to import graph in json format ( exported from gephi) but unsuccessful. Thank you in advance!

**Comment by timangus** [comment, marked as answer]
Supporting graph import in the fashion you describe is a feature we plan to implement soon. For the time being you can use pairwise text, which is essentially the same idea but instead of comma separating the values you use spaces, e.g.:

#### **`K33-pairwise.txt`**
```
"C" "F" 9
"C" "E" 8
"C" "D" 7
"B" "F" 6
"B" "E" 5
"B" "D" 4
"A" "F" 3
"A" "E" 2
"A" "D" 1
```

The trouble with JSON graph formats is that there are several of them (https://xkcd.com/927/) so you might have ended up with one we don't support. GML or GraphML tend to be more reliable as interchange formats, for what it's worth.

**Reply by NETfrix** [reply]
hi,
I tried to copy-paste your reply to a txt file and upload it but I got:
"The file cannot be loaded"

**Reply by timangus** [reply]
Not sure what to say, works fine here. What version are you running?

**Comment by NETfrix** [comment]
The latest... 4.1 right?
It's not the 1st time I had issues uploading data. What I finally did was to export a gml file from Gephi...
If you could publish a straightforward manual to upload an edge list (a simple "source-target" file) it would be highly appreciated!
Something like:


o upload a source-target file , you need to ensure that your data is in a specific format. Graphia primarily uses a TXT file format for importing source-target data. Here's how to format your source-target file correctly:

  1.  Header Row:

     *   The first row of your TXT file should contain column headers that describe your data. These headers should include at least two columns: one for the source nodes and another for the target nodes.
Example headers: "Source," "Target," and optionally, you can include other attributes as needed.

Data Rows:

     *   Each subsequent row in the TXT file represents an edge (connection) between source and target nodes.
The source and target columns should contain unique identifiers for each node. These identifiers could be names, IDs, or any other unique value.
Each row should specify the relationship between a source node and a target node.

Here's a simplified example of what your CSV file might look like:

Copy code
Source,Target
NodeA,NodeB
NodeA,NodeC
NodeB,NodeC
NodeD,NodeE


In this example, we have a network with five nodes (NodeA, NodeB, NodeC, NodeD, and NodeE) and four edges connecting them.

  1.  Optional Attributes:
     *   If you have additional attributes or properties associated with your nodes or edges (e.g., weight, time, category), you can include extra columns in your CSV file to represent these attributes. Make sure to include corresponding headers for these columns.

Here's an example with additional attributes:

mathematicaCopy code
Source,Target,Weight,CategoryNodeA,NodeB,0.5,RedNodeA,NodeC,0.8,BlueNodeB,NodeC,0.7,GreenNodeD,NodeE,0.4,Yellow

In this example, we've added two extra columns for "Weight" and "Category" attributes.



You're one of the few that allow people to engage with large networks. It's a pity that the uploading part gets in the way.

Thank you!
Best,
Asaf Shapira

________________________________
מאת: Tim Angus ***@***.***>
‏‏נשלח: יום שני 04 ספטמבר 2023 15:22
‏‏אל: graphia-app/graphia ***@***.***>
עותק: NETfrix ***@***.***>; Comment ***@***.***>
‏‏נושא: Re: [graphia-app/graphia] Data import (Discussion #37)


Not sure what to say, works fine here. What version are you running?

**Comment by timangus** [comment]
You're basically describing pairwise data, for which there is a loader: https://graphia.app/guide/section3/1_load_graph_data.html

**Comment by NETfrix** [comment]
I know! Weird, right?
This is what I got when I tried Xlsx/txt file:

[cid:0fa611e8-64a6-4121-a9f7-c1f25202ccc0]
________________________________
מאת: Tim Angus ***@***.***>
‏‏נשלח: יום שני 11 ספטמבר 2023 14:05
‏‏אל: graphia-app/graphia ***@***.***>
עותק: NETfrix ***@***.***>; Comment ***@***.***>
‏‏נושא: Re: [graphia-app/graphia] Data import (Discussion #37)


You're basically describing pairwise data, for which there is a loader: https://graphia.app/guide/section3/1_load_graph_data.html

**Comment by timangus** [comment]
> I know! Weird, right? This is what I got when I tried Xlsx/txt file: [cid:0fa611e8-64a6-4121-a9f7-c1f25202ccc0]

Not sure what you're trying to say here?

**Comment by NETfrix** [comment]
Sorry.
Here's the attached file that went missing on the previous mail.

________________________________
מאת: Tim Angus ***@***.***>
‏‏נשלח: יום שלישי 12 ספטמבר 2023 16:38
‏‏אל: graphia-app/graphia ***@***.***>
עותק: NETfrix ***@***.***>; Comment ***@***.***>
‏‏נושא: Re: [graphia-app/graphia] Data import (Discussion #37)


I know! Weird, right? This is what I got when I tried Xlsx/txt file: [cid:0fa611e8-64a6-4121-a9f7-c1f25202ccc0]

Not sure what you're trying to say here?
